packages/zendesk-adapter/src/filters/guide_theme.ts

```diff
@@ -13,41 +13,105 @@
+import {
+  AdditionChange,
+  Change,
+  getChangeData,
+  InstanceElement,
+  isAdditionOrModificationChange,
+  isInstanceElement,
+  isModificationChange,
+  isReferenceExpression,
+  ModificationChange,
+  SaltoElementError,
+  SaltoError,
+  StaticFile,
+  Element,
+  isObjectType, Field, MapType,
+} from '@salto-io/adapter-api'
 import { logger } from '@salto-io/logging'
+import { values } from '@salto-io/lowerdash'
 import JSZip from 'jszip'
 import _, { remove } from 'lodash'
-import { naclCase } from '@salto-io/adapter-utils'
-import { FETCH_CONFIG, isGuideEnabled, isGuideThemesEnabled } from '../config'
+import { getInstancesFromElementSource, naclCase } from '@salto-io/adapter-utils'
+import ZendeskClient from '../client/client'
+import { FETCH_CONFIG, isGuideThemesEnabled } from '../config'
 import {
-  GUIDE_THEME_TYPE_NAME, ZENDESK,
+  GUIDE_THEME_TYPE_NAME, THEME_FILE_TYPE_NAME, THEME_FOLDER_TYPE_NAME, THEME_SETTINGS_TYPE_NAME, ZENDESK,
 } from '../constants'
 import { FilterCreator } from '../filter'
+import { create } from './guide_themes/create'
+import { deleteTheme } from './guide_themes/delete'
 import { download } from './guide_themes/download'
+import { publish } from './guide_themes/publish'
 import { getBrandsForGuideThemes } from './utils'
 
 const log = logger(module)
 
 type ThemeFile = { filename: string; content: StaticFile }
-type ThemeDirectory = { [key: string]: ThemeFile | ThemeDirectory }
+type DeployThemeFile = { filename: string; content: Buffer }
+
 
-const unzipFolderToElements = async (buffer: Buffer, brandName: string, name: string): Promise<ThemeDirectory> => {
+type ThemeDirectory = {
+  files: Record<string, ThemeFile | DeployThemeFile>
+  folders: Record<string, ThemeDirectory>
+}
+
+const unzipFolderToElements = async (
+  buffer: Buffer, brandName: string, name: string,
+): Promise<ThemeDirectory> => {
   const zip = new JSZip()
   const unzippedContents = await zip.loadAsync(buffer)
 
-  const elements: ThemeDirectory = {}
+  const elements: ThemeDirectory = { files: {}, folders: {} }
+  const addFile = async (
+    fullPath: string,
+    pathParts: string[],
+    file: JSZip.JSZipObject,
+    currentDir: ThemeDirectory
+  ): Promise<void> => {
+    const [firstPart, ...rest] = pathParts
+
+    if (pathParts.length === 1) {
+      // It's a file
+      const filepath = `${ZENDESK}/themes/brands/${brandName}/${name}/${fullPath}`
+      const content = await file.async('nodebuffer')
+      currentDir.files[naclCase(firstPart)] = {
+        filename: fullPath,
+        content: new StaticFile({ filepath, content }),
+      }
+    } else {
+      // It's a folder
+      if (!currentDir.folders[naclCase(firstPart)]) {
+        currentDir.folders[naclCase(firstPart)] = { files: {}, folders: {} }
+      }
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      await addFile(fullPath, rest, file, currentDir.folders[naclCase(firstPart)])
+    }
```

Comment:
1. nit: can `return` early if there is nothing after the `else` and reduce nesting
```suggestion
      }
      return
    }
    // It's a folder
    if (!currentDir.folders[naclCase(firstPart)]) {
      currentDir.folders[naclCase(firstPart)] = { files: {}, folders: {} }
    }
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    await addFile(fullPath, rest, file, currentDir.folders[naclCase(firstPart)])
```
2. do we still need the `eslint-disable`?